php/WP_CLI/Bootstrap/IncludeRequestsAutoloader.php

```diff
@@ -48,16 +48,30 @@ public function process( BootstrapState $state ) {
 
 		$runner = new RunnerInstance();
 
-		// Make sure we don't deal with an invalid `--path` value.
-		$config = $runner()->config;
-		if ( isset( $config['path'] ) &&
-			( is_bool( $config['path'] ) || empty( $config['path'] ) )
-		) {
-			return $state;
+		// Use `--path` from the alias if one is matching.
+		$alias_path = null;
+		if ( $runner()->alias
+			&& isset( $runner()->aliases[ $runner()->alias ]['path'] ) ) {
+			$alias_path = $runner()->aliases[ $runner()->alias ]['path'];
+			// Make sure it isn't an invalid value.
+			if ( is_bool( $alias_path ) || empty( $alias_path ) ) {
+				return $state;
```

Comment:
Once we're here, is it guaranteed that `$runner()->find_wp_root()` cannot find a functional result? Because we're short-circuiting that here if the alias is not valid. So, if the alias is invalid, could the regular filesystem traversal still work? If yes, we cannot do an early return here.

Reply:
@schlessera I don't follow. This logic follows the same behavior as `find_wp_root()` when `--path=<path>` is provided:

https://github.com/wp-cli/wp-cli/blob/b6ab66710025d70e8975695a9f9ec2e51d69e6a1/php/WP_CLI/Runner.php#L259-L266

If the alias provides a `--path=<path>`, then we're explicitly checking it. If the alias doesn't provide a `--path=<path>`, then the `} else {` below is triggered and it looks for the WordPress root.

Does this clarify? If not, can you provide an example of what you're referring to?

Reply:
Before, we only checked `--path`. If it was invalid, we did an early return.

Now, we have two checks. We first check `$alias->path`, then we check regular `--path`. For both we do an early return if the value is invalid.

So what I'm wondering is whether the following scenario should work:
- we do have a value for both `$alias->path` and regular `--path`
- `$alias->path` is invalid
- `--path is valid

In that case, should we fall back to `--path` if `$alias->path` is invalid? If so, we cannot do an early return in the validity check for `$alias->path`, as that just skips the entire logic for `--path`.

Reply:
[Discussed in Slack](https://wordpress.slack.com/archives/C02RP4T41/p1694033733749739): early return is fine, this is a non-issue.